[jason-chuah] iP

jason-chuah opened the pull request:
# Park
> Too much light makes a baby blind -unknown

Park frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~very fast~~ very **very** fast

All you need to do is,
1. Download it from [here](https://github.com/jason-chuah/ip)
2. `run` Park.java and hope it works 🙃 

>[!NOTE]
>And it is ***FREE***!

Features:
- [x] Managing tasks
- [ ] Managing your life (coming soon)

Here is a very useful snippet of code:
```ruby
public static void main(String[] args) {
        new Park("./data/Park.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jason-chuah committed:
Change Duke.java

jason-chuah committed:
Edit for Level 1

jason-chuah committed:
Edit for Level 2

jason-chuah committed:
Edit for Level 3 and add Task.java

jason-chuah committed:
Edit for Level 4, added Deadline.java, Todo.java, Event.java

jason-chuah committed:
automate text UI testing

jason-chuah committed:
Implement error handling

jason-chuah committed:
Add delete functionality

jason-chuah committed:
Add save functionality

jason-chuah committed:
Merge branch 'branch-Level-7'

jason-chuah committed:
Add DateTime functionality

jason-chuah committed:
Merge branch 'branch-Level-8'

jason-chuah committed:
Refactor code to use more OOP

jason-chuah committed:
Merge branch 'A-MoreOOP'

jason-chuah committed:
Divide classes into packages

jason-chuah committed:
Merge branch 'A-Packages'

jason-chuah committed:
Merge branch 'add-gradle-support'

jason-chuah committed:
Add JUnit tests

jason-chuah committed:
Create JAR file

jason-chuah committed:
Add JavaDoc comments

jason-chuah committed:
Edit to comply to coding standards

jason-chuah committed:
Edit to comply to coding standards

jason-chuah committed:
Add find functionality

jason-chuah committed:
Merge branch 'A-JavaDoc'

jason-chuah committed:
Resolve conflicts

jason-chuah committed:
Merge branch 'Level-9'

jason-chuah committed:
Implement GUI

jason-chuah committed:
Merge branch 'branch-Level-10'

triishaa commented on src/main/java/park/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,42 @@
+            Task t = tasks.get(index);
+            storage.delete(t);
+            ui.setResponse("Noted. I've removed this task:" + t + "\n");
+            tasks.delete(index);
```
instead of having just 1 setResponse, would it be better to have different methods for different printing? 
easier to reuse. you'll have to type the response every time or else?

triishaa commented on src/main/java/park/commands/ExitCommand.java:
```diff
@@ -0,0 +1,21 @@
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ui.setResponse("Bye. Hope to see you again soon!");
```
check above comment

triishaa commented on src/main/java/park/commands/FindCommand.java:
```diff
@@ -0,0 +1,45 @@
+            }
+        }
+        if (result.isEmpty()) {
+            ui.setResponse("There are no matching tasks.");
```
check above comment

triishaa commented on src/main/java/park/commands/FindCommand.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
boolean isExit? I dont get the naming? is isExit correct?

triishaa commented on src/main/java/park/task/Event.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @Override
+    public String encode() {
```
maybe can use a bit more specific name? dateAfterEncoding? or something? encode seems too general

nayanaamahesh commented on src/main/java/park/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
```
I like how you organise the imports by separating different kinds of them

nayanaamahesh commented on src/main/java/park/Park.java:
```diff
@@ -0,0 +1,68 @@
+     */
+    public void run() {
+        ui.showWelcomeMessage();
+        boolean isExit = false;
```
I like how you named this variable

nayanaamahesh commented on src/main/java/park/storage/Storage.java:
```diff
@@ -0,0 +1,161 @@
+    public Task decode(String line) throws ParkException {
+        try {
+            String[] s = line.split("/");
+            switch (s[0]) {
```
good use of switch case

Shantaa reviewed (commented):
LGTM. I liked how you further split the commands into similar commands before working on their code.

Shantaa commented on src/main/java/park/parser/Parser.java:
I liked how you personalised each error so that the user knows what specifically went wrong. Any reason why you catch the respective errors and then throw a ParkException, instead of just using the error immediately?

jason-chuah committed:
Add Assertions to MarkCommand and UnmarkCommand

There is no verification for whether the Task.mark() and Task.unmark()
methods are executed correctly.

Verifying them ensures that the code is working as intended.

Let's add assertions so that the code does not execute with any
unexpected behaviour.

jason-chuah committed:
Enclose all response and error messages

Response and error messages are manually entered each time they are
invoked. This results in the same messages being entered multiple
times.

Removing such duplicates enhances code readability and better adheres
to OOP principles.

Let's enclose all messages shown to the user in the Ui and
ParkException classes as methods. This better suits their intended
purpose and makes subsequent usage of the same messages easier.

jason-chuah committed:
Merge pull request #1 from jason-chuah/branch-A-Assertions

Add Assertions to MarkCommand and UnmarkCommand

jason-chuah committed:
Merge branch 'master' into branch-A-CodeQuality

jason-chuah committed:
Merge pull request #2 from jason-chuah/branch-A-CodeQuality

Enclose all response and error messages

jason-chuah committed:
Add help command

jason-chuah committed:
Merge branch 'branch-C-Help'

jason-chuah committed:
Fix error regarding reading DateTime from storage

jason-chuah committed:
Add user guide

jason-chuah committed:
Update build.gradle to point to the correct main class

jason-chuah committed:
Fix a bug where having / in the description breaks the storage

jason-chuah committed:
Update README.md
